### ISSUE TITLE
Support auto indentation, just like `aggressive-indent-mode` in emacs?

### ISSUE THREAD

**Issue by DogLooksGood** [oakes/paren-soup#3]
when using parinfer, It's really difficult to edit from

```
(foo
   (bar)
   (baz))
```

to 

```
(xxx (foo
          (bar)
          (baz)))
```

with some stuff like `aggressive-indent-mode`, using parinfer will be more effective.

**Comment by DogLooksGood** [comment]
Parinfer has paren mode and indent mode, i think currently paren-soup use indent mode as default. Is there a way to switch to paren mode?

**Comment by oakes** [comment]
I finally worked on this :) I just released the preliminary work in version 2.8.0. If you have time, please try it out and let me know if it works the way you expect:

http://oakes.github.io/paren-soup/

**Comment by oakes** [comment]
I'll close this since it's done, but let me know if you run into any other issues with the current implementation.
